### PR TITLE
feat(cli): add description when choosing a commit rule

### LINKED ISSUE #1822: Add concise example when choosing a commit rule in `cz init`
### Description

When a new user initializes cz in a Git repository, they are prompted to choose a commit rule. However, it isn’t clear how each rule translates into an actual commit message. Providing a simple concise documentation a at this step would help clarify the differences and reduce user friction.

### Possible Solution

_No response_

### Additional context

```
? Please choose a supported config file:  pyproject.toml
? Please choose a cz (commit rule): (default: cz_conventional_commits) (Use arrow keys)
 » cz_conventional_commits
   cz_customize
   cz_jira <- should have documentation when hover over it
```

### Related issues

_No response_

### PR DESCRIPTION
**Pull request by namwoam**
Closes #1822 

## Description

This PR adds descriptions to the commit message rule selection step.

## Checklist

- [ ✅ ] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/contributing)

### Was generative AI tooling used to co-author this PR?

- [X] No

### Code Changes

- [] Add test cases to all the changes you introduce
- [✅ ] Run `uv run poe all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [✅ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [✅ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [ ] Run `uv run poe doc` locally to ensure the documentation pages renders correctly
- [ ] Check and fix any broken links (internal or external)

## Expected Behavior
When selecting the commit rule at `cz init`, it should appear a description when selecting a commit rule.

Before:
Screenshot 2026-01-31 at 11 08 56 AM

After:
Screenshot 2026-01-31 at 12 50 58 PM

## Additional Context

Issue: https://github.com/commitizen-tools/commitizen/issues/1822

### CODE REVIEW

File: commitizen/commands/init.py [hunk 1]
```diff
@@ -165,9 +168,25 @@ def _ask_config_path(self) -> Path:
...
+        def construct_choice_with_description(cz_name: str) -> questionary.Choice:
+            try:
+                cz_class = registry.get(cz_name)
+                if cz_class:
+                    cz_obj = cz_class(self.config)
+                    first_example = cz_obj.schema().partition("\n")[0]
+                    return questionary.Choice(
+                        title=cz_name, value=cz_name, description=f"{first_example}"
+                    )
+            except Exception:  # pylint: disable=broad-except
```
Comment (bearomorphism) [inline commitizen/commands/init.py]: quick question: what are possible exceptions

the `try:` block should be as small as possible

File: commitizen/commands/init.py [hunk 2]
```diff
@@ -165,9 +168,25 @@ def _ask_config_path(self) -> Path:
...
 
     def _ask_name(self) -> str:
+        def construct_choice_with_description(cz_name: str) -> questionary.Choice:
+            try:
+                cz_class = registry.get(cz_name)
+                if cz_class:
+                    cz_obj = cz_class(self.config)
+                    first_example = cz_obj.schema().partition("\n")[0]
+                    return questionary.Choice(
+                        title=cz_name, value=cz_name, description=f"{first_example}"
```
Comment (bearomorphism) [inline commitizen/commands/init.py]:
```suggestion
                        title=cz_name, value=cz_name, description=first_example
```

File: commitizen/commands/init.py (hunk 1 above)
Comment (namwoam) [reply commitizen/commands/init.py]: cz_obj = cz_class(self.config) could cause exception `MissingCzCustomizeConfigError` for `cz_customize`, fixed except statement

File: commitizen/commands/init.py (hunk 2 above)
Comment (namwoam) [reply commitizen/commands/init.py]: Fixed

File: commitizen/commands/init.py [hunk 3]
```diff
@@ -165,9 +169,25 @@ def _ask_config_path(self) -> Path:
         return Path(filename)
 
     def _ask_name(self) -> str:
+        def construct_choice_with_description() -> list[questionary.Choice]:
```
Comment (Lee-W) [inline commitizen/commands/init.py]: do we need to make it a function inside `_ask_name`?

Comment (namwoam) [reply commitizen/commands/init.py]: Added unittest for construct_choice_with_description
